bindings/c/test/mako/mako.cpp

```diff
@@ -0,0 +1,2380 @@
+	                                        { "INSERTRANGE", { StepKind::IMM }, true },
+	                                        { "OVERWRITE", { StepKind::IMM }, true },
+	                                        { "CLEAR", { StepKind::IMM }, true },
+	                                        { "SETCLEAR", { StepKind::COMMIT, StepKind::IMM }, true },
+	                                        { "CLEARRANGE", { StepKind::IMM }, true },
+	                                        { "SETCLEARRANGE", { StepKind::COMMIT, StepKind::IMM }, true },
+	                                        { "COMMIT", { StepKind::NONE }, false },
+	                                        { "TRANSACTION", { StepKind::NONE }, false },
+	                                        { "READBLOBGRANULE", { StepKind::ON_ERROR }, false } } };
+
+const std::map<std::pair<int /*op*/, int /*sub-op step*/>,
+               Future (*)(TX, mako_args_t const&, ByteString& /*key1*/, ByteString& /*key2*/, ByteString& /*value*/)>
+    operation_fn_table{
+	    { { OP_GETREADVERSION, 0 },
+	      [](TX tx, mako_args_t const&, ByteString&, ByteString&, ByteString&) {
+	          return tx.get_read_version().erase_type();
+	      } },
+	    { { OP_GET, 0 },
+	      [](TX tx, mako_args_t const& args, ByteString& key, ByteString&, ByteString&) {
+	          const auto num = next_key(args);
+	          genkey(key, KEY_PREFIX, args, num);
+	          return tx.get(key, false /*snapshot*/).erase_type();
+	      } },
+	    { { OP_GETRANGE, 0 },
+	      [](TX tx, mako_args_t const& args, ByteString& begin, ByteString& end, ByteString&) {
+	          const auto num_begin = next_key(args);
+	          genkey(begin, KEY_PREFIX, args, num_begin);
+	          auto num_end = num_begin + args.txnspec.ops[OP_GETRANGE][OP_RANGE] - 1;
+	          if (num_end > args.rows - 1)
+		          num_end = args.rows - 1;
+	          genkey(end, KEY_PREFIX, args, num_end);
+	          return tx
+	              .get_range<key_select::inclusive, key_select::inclusive>(begin,
+	                                                                       end,
+	                                                                       0 /*limit*/,
+	                                                                       0 /*target_bytes*/,
+	                                                                       args.streaming_mode,
+	                                                                       0 /*iteration*/,
+	                                                                       false /*snapshot*/,
+	                                                                       args.txnspec.ops[OP_GETRANGE][OP_REVERSE])
+	              .erase_type();
+	      } },
+	    { { OP_SGET, 0 },
+	      [](TX tx, mako_args_t const& args, ByteString& key, ByteString&, ByteString&) {
+	          const auto num = next_key(args);
+	          genkey(key, KEY_PREFIX, args, num);
+	          return tx.get(key, true /*snapshot*/).erase_type();
```

Comment:
For completeness of benchmarking we should also actually retrieve the returned values for read operations, i.e. calling fdb_future_get_value or fdb_future_get_keyvalue_array, etc. on the returned future

Reply:
Done!